bots/task/__init__.py

```diff
@@ -268,7 +269,26 @@ def run(context, function, **kwargs):
         finish(publishing, ret, name, context, issue)
     return ret or 0
 
-def stale(days, pathspec):
+def checkout(ref="HEAD"):
+    if ref:
+        execute("git", "checkout", "--detach", ref)
+
+    # COMPAT: If the bots directory doesn't exist in this branch, check it out from master
+    if subprocess.call([ "git", "ls-tree", "-d", "HEAD:bots/"], stdout=DEVNULL) != 0:
+        sys.stderr.write("Checking out bots directory from master ...\n")
+        subprocess.check_call([ "git", "checkout", "--force", "origin/master", "--", "bots/" ])
+
+        # The machine code is special copy it from master too
+        machine = os.path.join(BOTS, "machine")
+        for name in os.listdir(machine):
+            path = os.path.join(machine, name)
+            if os.path.islink(path):
+                os.unlink(path)
+                code = subprocess.check_output([ "git", "show", "origin/master:test/common/{0}".format(name) ])
+                with open(path, "w") as f:
+                    f.write(code)
+
+def stale(days, pathspec, ref="HEAD"):
```

Comment:
While we're touching this, it would be nice to add a comment here.
At the very least it'd be good to know that this function only checks something and doesn't communicate with GitHub et cetera. Or is this a follow-up?